code/modules/spells/spell_types/mime.dm

```diff
@@ -59,4 +59,74 @@
 		if(H.mind.miming)
 			H << "<span class='notice'>You make a vow of silence.</span>"
 		else
-			H << "<span class='notice'>You break your vow of silence.</span>"
\ No newline at end of file
+			H << "<span class='notice'>You break your vow of silence.</span>"
+
+// The spell can only be gotten from the "Advanced Mimery Kit" for Mime Traitors.
+
+/obj/effect/proc_holder/spell/aoe_turf/conjure/mime_wall/blockade
+	name = "Invisible Blockade"
+	desc = "With more polished skills, a powerful mime can create a invisble blockade, blocking off a 3x3 area."
+	summon_type = list(/obj/effect/forcefield/mime/advanced)
+	invocation_type = "emote"
+	invocation_emote_self = "<span class='notice'>You form a blockade in front of yourself.</span>"
+	summon_lifespan = 600
+	charge_max = 600
+	summon_amt = 9
+	summon_ignore_prev_spawn_points = 1
+	range = 1
+
+/obj/effect/proc_holder/spell/aoe_turf/conjure/mime_wall/blockade/Click()
+	if(usr && usr.mind)
+		if(!usr.mind.miming)
+			usr << "<span class='notice'>You must dedicate yourself to silence first.</span>"
+			return
+		invocation = "<B>[usr.real_name]</B> looks as if a blockade is in front of [usr.p_them()]."
+	else
+		invocation_type ="none"
+	..()
+
+/obj/effect/proc_holder/spell/aimed/finger_guns
+	name = "Finger Guns"
+	desc = "An ancient technqiue, passed down from mentor to student. Allows you to shoot bullets out of your fingers."
+	school = "mime"
+	panel = "Mime"
+	charge_max = 300
+	clothes_req = 0
+	invocation_type = "emote"
+	invocation_emote_self = "<span class='dangers'>You fire your finger gun!</span>"
+	range = 20
+	projectile_type = /obj/item/projectile/bullet/weakbullet2
+	sound = null
+	active_msg = "You draw your fingers!"
+	deactive_msg = "You put your fingers at ease. Another time."
+	active = FALSE
+
+	action_icon_state = "mime"
+	action_background_icon_state = "bg_mime"
+	base_icon_state = "mime"
+
+
+/obj/effect/proc_holder/spell/aimed/finger_guns/Click()
+	if(usr && usr.mind)
+		if(!usr.mind.miming)
+			usr << "<span class='notice'>You must dedicate yourself to silence first.</span>"
+			return
+		invocation = "<B>[usr.real_name]</B> pulls the finger gun's trigger!."
+	else
+		invocation_type ="none"
+	..()
+
+
+/obj/item/weapon/spellbook/oneuse/mimery_one

```

Comment:
sure but I also like to adminspawn spellbook/random a lot and this would fall under that